tests/workflows/test_workflow_test_helpers.py

```
@@ -0,0 +1,167 @@
+
+
+class TestCargoMetadata:
+    """Tests for `_cargo_metadata()` error handling and edge cases."""
+
+    def test_requires_cargo_on_path(self) -> None:
+        """The helper fails fast when `cargo` cannot be resolved."""
+        with patch("tests.workflows.workflow_test_helpers.shutil.which", return_value=None):
+            with pytest.raises(
+                AssertionError,
+                match="cargo executable must be available in PATH",
+            ):
```

Ruff (0.15.9)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)